neuralpp/experiments/experimental_inference/exact_belief_propagation.py

```diff
@@ -92,9 +93,13 @@ def expand_partial_tree_and_recompute(self, expansion_root):
         potential_expansion = self.expansion[expansion_root]
         if potential_expansion is None:
             return
-        self.expansion.expand_partial_tree_and_recompute(expansion_root)
-        self.recompute_and_propagate_result_to_ancestors(potential_expansion.node)
+        expand_to_node = potential_expansion.node
+        parent = self.full_tree.parent(expand_to_node)
+
+        self.tree.add_edge(parent, expand_to_node)
```

Comment:
If we have a super class `PartialTreeComputation`, then adding an edge to it would be all that's needed at this point. The value updating mechanics would happen there.